cpp/ql/src/semmle/code/cpp/models/implementations/StdMap.qll

```diff
@@ -67,6 +112,22 @@ class StdMapSwap extends TaintFunction {
   }
 }
 
+/**
+ * The standard map `merge` function.
+ */
+class StdMapMerge extends TaintFunction {
+  StdMapMerge() { this.hasQualifiedName("std", ["map", "unordered_map"], "merge") }
+
+  override predicate hasTaintFlow(FunctionInput input, FunctionOutput output) {
+    // container1.merge(container2)
+    input.isQualifierObject() and
+    output.isParameterDeref(0)
```

Comment:
I think the only taint that would propagate from the qualifier to the parameter is from keys, not values, so we don't want taint here (until we have separate taint through keys).

Reply:
Good point, the merge isn't symmetric.  Fixed.

Even for keys the taint in that direction would be dubious.  The only thing the keys in the qualifier control is whether or not entries are removed from the parameter, and generally we don't consider boolean decision making (such as `if`) to propagate taint.